Финальный проект 6 спринта

Julia13179 opened the pull request.

Julia13179 committed:
Финальный проект 6 спринта

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,39 @@
+    service = Service()
+    drv = webdriver.Firefox(service=service, options=opt)
+    drv.set_window_size(1280, 1024)
+    drv.implicitly_wait(5)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions и встраивать ожидания на нем.

Julia13179 replied:
исправлено

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,22 @@
+BASE_URL = 'https://qa-scooter.praktikum-services.ru/'
+ORDER_PAGE_URL = 'https://qa-scooter.praktikum-services.ru/order'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов. Записать в переменную baseUrl и прикручивать к ней эндпоинты

Julia13179 replied:
исправлено

AlexKitIt commented on pages/order_form_page.py:
```diff
@@ -0,0 +1,106 @@
+    @allure.step('Заполнить поле "Имя"')
+    def set_first_name(self, name):
+        self.set_text_to_elm(L.FIRST_NAME_FIELD, name)
+        return self
```
Нужно исправить: здесь и далее: не нужно тут возвращать self. писать цепочки вызовов неинформативно и очень сложно поддается дебагу

Julia13179 replied:
исправлено

AlexKitIt commented on pages/order_form_page.py:
```diff
@@ -0,0 +1,106 @@
+        
+        self.click_on_element(L.METRO_STATION_FIELD)
+        self.set_text_to_elm(L.METRO_STATION_FIELD, station)
+        WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(L.METRO_STATION_LIST))
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к  WebDriverWait необходимо вынести в base, там описать методы взаимодействия с ними и оттуда за счет наследования классов страниц от base переиспользовать

Julia13179 replied:
исправлено

AlexKitIt commented on tests/test_home_page.py:
```diff
@@ -0,0 +1,39 @@
+    @allure.description('Проверка открытия страницы Яндекс.Дзен в соседней вкладке при нажатии на логотип "Яндекс"')
+    def test_clicking_yandex_logo_opens_dzen_page(self, driver, home_page):
+        home_page.click_logo_yandex_open_dzen_page()
+        assert driver.current_url == DZEN_URL
```
Нужно исправить: здесь и далее: прямого обращения к driver, вызова его property, execute_script на нем и тд из тестов быть не должно. Для этого в base пишутся соответствующие методы и тут вызываются на объектах конкретных страниц

Julia13179 replied:
исправлено

AlexKitIt commented on tests/test_order_form_page.py:
```diff
@@ -0,0 +1,31 @@
+    ("header", user_1),
+    ("body",   user_2),
+])
+def test_order_form_flow(home_page, order_page, start_button, user):
```
Нужно исправить: тестовые методы в модулях объединяются в тестовые классы, необходимо создать в каждом тестовом модуле тестовые классы и разместить в них методы, даже если тестовый метод только один

Julia13179 replied:
исправлено

AlexKitIt commented on tests/test_order_form_page.py:
```diff
@@ -0,0 +1,31 @@
+    name, last_name, address, station, number, comment = user
+
+    # открыть форму
+    if start_button == "header":
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен. Если это не вписывается в параметризацию, значит ее быть не должно и это просто несколько отдельных тестовых методов

Julia13179 replied:
исправлено

AlexKitIt commented on tests/test_order_form_page.py:
```diff
@@ -0,0 +1,31 @@
+    order_page.click_yes_button_confirmation_pop_up()
+
+    # проверить
+    assert order_page.find_element_with_wait(L.POP_UP_COMPLETE_ORDER).is_displayed(), \
```
Нужно исправить: здесь и далее: методов с принимаемыми локаторами в тестах быть не должно, если это не вариант параметризации, они живут в методах пейджи. В тесте только создание объекта и вызов метода, без указания локаторов элементов с которым происходит взаимодействие. Тест должен быть максимально изолирован от логики выполнения

Commit:
Исправлены замечания ревьюера

- Убраны неявные ожидания из conftest.py
- Исправлены URL в data.py для конкатенации с базовым URL
- Вынесены WebDriverWait в base класс для переиспользования
- Добавлены методы для работы с ожиданиями в base_page.py

Commit:
Завершены исправления замечаний ревьюера

- Убраны возвраты self из методов страниц (цепочки вызовов)
- Убраны прямые обращения к driver из тестов
- Созданы тестовые классы в каждом тестовом модуле
- Убраны условия из тестов, разделены на отдельные методы
- Убраны методы с локаторами из тестов, перенесены в page objects
